crates/sui-kvstore/src/bigtable/client.rs

```diff
@@ -423,6 +434,47 @@ impl BigTableClient {
         &mut self,
         table_name: &str,
         keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<Vec<(Bytes, Bytes)>>> {
+        let elapsed = Instant::now().elapsed();
+        let num_keys_requested = keys.len();
+        let result = self.multi_get_internal(table_name, keys).await;
+        let labels = [&self.client_name, table_name];
+        match &self.metrics {
+            None => result,
+            Some(metrics) => match result {
+                Err(e) => {
+                    metrics.kv_get_errors.with_label_values(&labels).inc();
+                    Err(e)
+                }
+                Ok(result) => {
+                    metrics
+                        .kv_get_batch_size
+                        .with_label_values(&labels)
+                        .observe(num_keys_requested as f64);
+                    if num_keys_requested > result.len() {
+                        metrics
+                            .kv_get_not_found
+                            .with_label_values(&labels)
+                            .inc_by((num_keys_requested - result.len()) as u64);
+                    }
+                    metrics
+                        .kv_get_success
+                        .with_label_values(&labels)
+                        .inc_by(result.len() as u64);
+                    metrics
+                        .kv_get_latency_ms
```

Comment:
Can we add a histogram metrics `kv_get_latency_ms_per_key` that is basically `kv_get_latency_ms` divided by `num_keys_requested`? so we know if something is slow due to large size or something else.